### PR TITLE
p2p: refactor and add checks to p2p tests

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes

This pull request refactors and adds checks to the p2p package tests.

## Tests:
```
go test ./p2p -v
```

### CODE REVIEW

File: p2p/p2p_test.go [hunk 1]
```diff
@@ -17,25 +17,18 @@
...
 	"github.com/ChainSafe/gossamer/common/optional"
-	net "github.com/libp2p/go-libp2p-core/network"
 	peer "github.com/libp2p/go-libp2p-core/peer"
-	ps "github.com/libp2p/go-libp2p-core/peerstore"
-	ma "github.com/multiformats/go-multiaddr"
 )
 
-func startNewService(t *testing.T, cfg *Config, msgSendChan chan []byte, msgRecChan chan BlockAnnounceMessage) *Service {
-	node, err := NewService(cfg, msgSendChan, msgRecChan)
+func startNewService(t *testing.T, cfg *Config, sendChan chan []byte, recChan chan BlockAnnounceMessage) *Service {
```
Comment (ansermino) [inline p2p/p2p_test.go]: I think you removed the wrong part, having `chan` in the name is a cardinal sin.

File: p2p/p2p_test.go [hunk 2]
```diff
@@ -48,407 +41,337 @@ func startNewService(t *testing.T, cfg *Config, msgSendChan chan []byte, msgRecC
...
-	nodeCfg := &Config{
-		BootstrapNodes: []string{bootnodeAddr.String()},
-		Port:           7001,
+	configB := &Config{
+		BootstrapNodes: []string{addrA.String()},
+		Port:           7002,
 		RandSeed:       2,
-		NoBootstrap:    false,
-		NoMdns:         true,
+		NoMdns:         true, // TODO: investigate failed dials, disable for now
```
Comment (ansermino) [inline p2p/p2p_test.go]: IMO repeating this everywhere is unnecessary work to remove, opening an issue would be sufficient.

File: p2p/config_test.go [hunk 3]
```diff
@@ -1,41 +1,130 @@
...
+	configA := &Config{
+		DataDir: testDir,
+	}
+
+	_, err := configA.buildOpts()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if configA.BootstrapNodes != nil {
```
[the remark this answers is not in the document]
Comment (ryanchristo) [reply p2p/config_test.go]: Good idea. Updated.

File: p2p/p2p_test.go (hunk 1 above)
Comment (ryanchristo) [reply p2p/p2p_test.go]: Ha! Noted and updated to `msgSend` and `msgRec`.

File: p2p/p2p_test.go (hunk 2 above)
Comment (ryanchristo) [reply p2p/p2p_test.go]: Yea, maybe a bit excessive but the intention was to make it clear where to remove when the issue is resolved. Can we keep it for now?

File: p2p/p2p_test.go [hunk 4]
```diff
@@ -48,407 +41,337 @@ func startNewService(t *testing.T, cfg *Config, msgSendChan chan []byte, msgRecC
...
 
-	msgChanC := make(chan []byte)
-	nodeC := startNewService(t, nodeConfigC, msgChanC, nil)
+	msgSendC := make(chan []byte)
+
+	nodeC := startNewService(t, configC, msgSendC, nil)
 	defer nodeC.Stop()
 
-	// Meaningless hash
+	// Create end block hash (hex has no significance here)
```
Comment (noot) [inline p2p/p2p_test.go]: I wouldn't say hex has no significance here, it's more like this hash is arbitrary

Comment (ryanchristo) [reply p2p/p2p_test.go]: Good point. :+1:

File: p2p/p2p_test.go (hunk 2 above)
Comment (ansermino) [reply p2p/p2p_test.go]: Fine to leave as you will surely be the one removing them :)

**Pull request merged**